utils/fitbit.js

```diff
@@ -12,17 +13,31 @@ module.exports = exports = {
       userAuthorizationURL: 'https://www.fitbit.com/oauth/authorize'
     }, function (token, tokenSecret, profile, done) {   
       //after oath login call this success handler
-          //add user to db
-          dbHelper.addUser(token, tokenSecret, profile);  
-          //this line waits for 26 to finish
-          exports.getStats(profile.id, token, tokenSecret).then(function() { 
-            //done tells the program you are done and want to go to the next step
-            done(null, profile._json.user); 
+      // add or update user in db
+      dbHelper.addUser(token, tokenSecret, profile, function(user) {
+        // user contains user db model
+        // update user fitbit stats in db
+        // console.log('addUser finished', user);
+        exports.getStats(user.id, token, tokenSecret).then(function() { 
```

Comment:
can you have an export inside a callback like that? not sure i've ever seen that before